### DISCUSSION TITLE
Where to store json files with site properties to be used across the code ?

### DISCUSSION THREAD

**Discussion by narasimhajupally** [nuxt/nuxt#18765] [Questions] [answered]
I am porting a next js open source template to nuxt.
In that project, they used files like
`data/headerNavLinks.js`  (to render nav links)
`data/siteMetadata.js` 
which are present in the `data` folder and are exporting jsons.
for example, siteMetadata json has
```
const siteMetadata = {
  title: 'Next.js Starter Blog',
  author: 'Tails Azimuth',
  headerTitle: 'TailwindBlog',
  description: 'A blog created with Next.js and Tailwind.css',
  language: 'en-us',
  ... so many other properties,
 comment: {
    provider: 'giscus',
    giscusConfig: {
      repo: process.env.NEXT_PUBLIC_GISCUS_REPO, //has access to env variables
      repositoryId: process.env.NEXT_PUBLIC_GISCUS_REPOSITORY_ID,
      category: process.env.NEXT_PUBLIC_GISCUS_CATEGORY,
      categoryId: process.env.NEXT_PUBLIC_GISCUS_CATEGORY_ID,
      lang: 'en',
    },
  ..etc
}
```
as we can see, it also has properties reading values from the env file.
So what is the proper way of implementing this in nuxt 3? so that where ever title or any other data is required I can access it using `siteMetadata.title` etc

**Comment by TheAlexLichter** [comment, marked as answer]
Instead of a JSON file, you could use the [app config](https://nuxt.com/docs/guide/directory-structure/app-config) or [runtime config](https://nuxt.com/docs/guide/going-further/runtime-config). The later especially for env variable support.

**Comment by narasimhajupally** [comment]
Thanks for the reply.
I was thinking the same and as for nav links I can store them in the nav component itself.
